test/hotspot/jtreg/vmTestbase/nsk/jdi/ThreadReference/stop/stop002.java

```diff
@@ -101,75 +112,196 @@ private int runIt(String args[], PrintStream out) {
                 + DEBUGGEE_THRNAME);
             tot_res = Consts.TEST_FAILED;
             return quitDebuggee();
         }
 
-        Field doExit = null;
+        Field stopLoop1 = null;
+        Field stopLoop2 = null;
         ObjectReference objRef = null;
+        ObjectReference throwableRef = null;
         try {
             // debuggee main class
-            ReferenceType rType = debuggee.classByName(DEBUGGEE_CLASS);
+            mainClass = debuggee.classByName(DEBUGGEE_CLASS);
 
-            suspendAtBP(rType, DEBUGGEE_STOPATLINE);
-            objRef = findObjRef(thrRef, DEBUGGEE_LOCALVAR);
+            suspendAtBP(mainClass, DEBUGGEE_STOPATLINE);
+            objRef = findObjRef(thrRef, DEBUGGEE_NON_THROWABLE_VAR);
+            throwableRef = findObjRef(thrRef, DEBUGGEE_THROWABLE_VAR);
 
-            // this field is used to indicate that debuggee has to
-            // stop looping
-            doExit = rType.fieldByName(DEBUGGEE_FIELD);
-
-            log.display("Resuming debuggee VM ...");
-            vm.resume();
-            log.display("Resumption of debuggee VM done");
+            // These fields are used to indicate that debuggee has to stop looping
+            stopLoop1 = mainClass.fieldByName(DEBUGGEE_STOP_LOOP1_FIELD);
+            stopLoop2 = mainClass.fieldByName(DEBUGGEE_STOP_LOOP2_FIELD);
+            if (stopLoop1 == null || stopLoop2 == null) {
+                throw new RuntimeException("Failed to find a \"stop loop\" field");
+            }
 
-            log.display("\nTrying to stop debuggee thread \"" + thrRef
-                + "\"\n\tusing non-throwable object reference \""
-                + objRef + "\" as a parameter ...");
+            log.display("non-throwable object: \"" + objRef + "\"");
+            log.display("throwable object:     \"" + throwableRef + "\"");
+            log.display("debuggee thread:      \"" + thrRef + "\"");
 
-// Check the tested assersion
+            /*
+             * Test #1: verify using a non-throwable object with stop() fails appropriately.
+             */
+            log.display("\nTEST #1: Trying to stop debuggee thread using non-throwable object.");
             try {
-                thrRef.stop(objRef);
-                log.complain("TEST FAILED: expected IllegalArgumentException was not thrown"
-                    + "\n\twhen attempted to stop debuggee thread \"" + thrRef
-                    + "\"\n\tusing non-throwable object reference \""
-                    + objRef + "\" as a parameter");
+                thrRef.stop(objRef); // objRef is an instance of the debuggee class, not a Throwable
+                log.complain("TEST #1 FAILED: expected IllegalArgumentException was not thrown");
                 tot_res = Consts.TEST_FAILED;
-            } catch(InvalidTypeException ee) {
-                log.display("CHECK PASSED: caught expected " + ee);
-            } catch(Exception ue) {
+            } catch (InvalidTypeException ee) {
```

Comment:
It is confusing that the log.complain at line 152 expects `IllegalArgumentException` but the really expected exception is `InvalidTypeException`. Is it intentional? If so then it is better to clarify it in the comment.

Reply:
Yes, this looks like a pre-existing bug. I'll cleanup the message.

Reply:
Thank you for the update!